src/main/java/com/flagsmith/threads/RequestProcessor.java

```
@@ -0,0 +1,107 @@
+    this.client = client;
+    this.logger = logger;
+    this.retries = retries;
+  }
+
+  /**
+   * Execute the request in async mode.
+   * @param request request to invoke
+   * @param clazz class type of response
+   * @param doThrow should throw Exception (boolean)
+   * @param <T> Type inference for the response
+   * @return
+   */
+  public <T> Future<T> executeAsync(Request request, TypeReference<T> clazz, Boolean doThrow) {
+    return executeAsync(request, clazz, doThrow, retries);
+  }
+
+  /**
+   * Execute the response in async mode.
+   * @param request Request object
+   * @param clazz class type of response
+   * @param doThrow should throw Exception
+   * @param retries no of retries before failing
+   * @param <T> Type inference for the response
+   * @return
+   */
+  public <T> Future<T> executeAsync(
+      Request request, TypeReference<T> clazz, Boolean doThrow, Integer retries) {
+    CompletableFuture<T> completableFuture = new CompletableFuture<>();
+    Call call = getClient().newCall(request);
+    // run the execute method in a fixed thread with retries.
+    executor.submit(() -> {
+      Integer localRetries = retries;
+      // retry until local retry reaches 0
+      try {
+        while (localRetries > 0) {
+          Boolean throwOrNot = localRetries == 1 ? doThrow : Boolean.FALSE;
+          try (Response response = call.execute()) {
+            if (response.isSuccessful()) {
+              ObjectMapper mapper = MapperFactory.getMappper();
+              completableFuture.complete(mapper.readValue(response.body().string(), clazz));
+              // break the while
+              break;
+
+            } else {
+              getLogger().httpError(request, response, throwOrNot);
+            }
+          } catch (IOException e) {
+            getLogger().httpError(request, e, throwOrNot);
+          }
+
+          localRetries--;
+        }
+      } catch (FlagsmithException e) {
+        // ignore this exception
```

This is probably my lack of understanding but I'm not sure why we'd want to swallow this error here? I can't actually see from the code here what might throw this exception in the first place 🤔